sqlglot/parser.py

```diff
@@ -2849,21 +2843,23 @@ def _parse_function(
             function = functions.get(upper)
             args = self._parse_csv(self._parse_lambda)
 
-            if function:
-                # Clickhouse supports function calls like foo(x, y)(z), so for these we need to also parse the
-                # second parameter list (i.e. "(z)") and the corresponding function will receive both arg lists.
-                if count_params(function) == 2:
-                    params = None
-                    if self._match_pair(TokenType.R_PAREN, TokenType.L_PAREN):
-                        params = self._parse_csv(self._parse_lambda)
-
-                    this = function(args, params)
+            # ClickHouse supports parametric function calls like `foo(param1, param2)(arg1, arg2)`
+            if self._match_pair(TokenType.R_PAREN, TokenType.L_PAREN):
```

Comment:
I'm not sure if this is correct, because e.g. in Spark you can have

```sql
select posexplode(array(5,6,7)) (p,c)
```

You don't want to parse this as a parametric function, `(p, c)` is an alias expression without the `AS` keyword.

Reply:
Of course...